org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java

```diff
@@ -398,12 +417,23 @@ private boolean runIlastik(final String tmpDirPath, final List<String> inFiles)
      * @param in input stream
      * @throws IOException
      */
-    static void writeToKnimeConsole(final InputStream in) throws IOException {
-        BufferedReader bis = new BufferedReader(new InputStreamReader(in, Charset.defaultCharset()));
-        String line;
-        while ((line = bis.readLine()) != null) {
-            KNIPGateway.log().info(line);
-        }
+    static void redirectToKnimeConsole(final InputStream in) {
+
+        new Thread(
+                   new Runnable() {
+                       @Override
+                       public void run() {
+                           BufferedReader bis = new BufferedReader(new InputStreamReader(in, Charset.defaultCharset()));
+                           String line;
+                           try {
+                               while ((line = bis.readLine()) != null) {
+                                   KNIPGateway.log().debug(line);
+                               }
+                           } catch (IOException ioe) {
+                               throw new RuntimeException("Could not read ilastik output", ioe); //FIXME not sure if this is a good idea
```

Comment:
what about that RuntimeException? I don't really like it, but I don't know if there's a more Knime-ish way to handle this